src/components/ui/Tiles/Tiles.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { forwardRef } from 'react'
+import type { HTMLAttributes, ReactElement } from 'react'
+
+import Tile from './Tile'
+
+export interface TilesProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+}
+
+const MIN_CHILDREN = 2
+const MAX_CHILDREN = 4
+const NUMBER_ITEMS_TO_EXPAND = 3
+
+const Tiles = forwardRef<HTMLDivElement, TilesProps>(function Tiles(
+  { testId = 'store-tiles', children, ...otherProps },
+  ref
+) {
+  const childrenCount = React.Children.count(children)
+  const spanFirstChild = childrenCount === NUMBER_ITEMS_TO_EXPAND
+  const isOutOfBounds =
+    childrenCount < MIN_CHILDREN || childrenCount > MAX_CHILDREN
+
+  if (isOutOfBounds) {
+    throw new Error(
+      `Tiles cannot receive less than ${MIN_CHILDREN} or more than ${MAX_CHILDREN} children.`
+    )
+  }
+
+  React.Children.forEach(children as ReactElement, (child) => {
+    if (child.type !== Tile) {
+      throw new Error('Only Tile components allowed as children.')
+    }
+  })
+
+  return (
+    <div ref={ref} data-store-tiles data-testid={testId} {...otherProps}>
+      {React.Children.map(children as ReactElement, (child, index) => {
+        if (index === 0 && spanFirstChild) {
+          const { className } = child.props
+
+          return React.cloneElement(child, {
+            className: className
+              ? `${className} tile-item-spanded`
```

Comment:
I'm my opinion the children should be opaque for the parent. You can implement this using another approach via CSS.

```tsx
return (
  <div {...otherProps} data-store-tiles={spanFirstChild ? 'expanded' : 'normal'}>{children}</div>
)
```

on the CSS file you can do this:
```css
[data-store-tiles] {
  /* whatever CSS you wan on both cases normal or expanded */
}

[data-store-tiles= "expanded"] > [data-store-tile]:first-child {
  /* whatever CSS you want to apply to the first data-store-tile element **when** data-store-tiles is expanded */
}
```

Reply:
It will make your code a little bit faster and easier to read.

Reply:
Heey, I really liked this approach! I'm gonna implement it, thanks!